### ISSUE TITLE
false negative when adding not null column

### ISSUE THREAD

**Issue by mockdeep** [ankane/strong_migrations#213]
For some reason, when we have the following migration StrongMigrations does not report an issue, even though it seems like it should complain about adding `NOT NULL`:

```rb
class AddSettingsToCampaigns < ActiveRecord::Migration[7.0]

  def change
    add_column :campaigns, :settings, :jsonb, null: false, default: {}
  end

end
```

It's been blocked by `statement_timeout` after several attempts to run it on our production, so I'm guessing it's the addition of the null check that's causing issues.

**Comment by ankane** [comment]
Hey @mockdeep, that operation should be safe.

```sql
CREATE TABLE campaigns (value int);
INSERT INTO campaigns SELECT i FROM generate_series(1,10000000) i;
ALTER TABLE "campaigns" ADD "settings" jsonb DEFAULT '{}' NOT NULL; -- fast: 20 ms
```

If you can create a script to reproduce, feel free to open a new issue.

**Comment by ankane** [comment]
fwiw, it sounds more like a lock timeout (but could show up as a statement timeout if the statement timeout is lower than the lock timeout)

**Comment by mockdeep** [comment]
Yeah, it's the lock timeout we're hitting. The statement timeout is currently set at 2 hours.

**Comment by ankane** [comment]
That's not uncommon for busy tables (and not considered dangerous), so don't think there's anything to do here.
